### PR TITLE
feat: use asset resolution in `kili.export_labels`

### PR DESCRIPTION
**Pull request by Jonas1312**

### CODE REVIEW

**Review by PierreLeveau — APPROVED** [review]
Approved (but do not release before next Kili release)

File: src/kili/services/export/tools.py [hunk 1]
```diff
@@ -164,3 +161,25 @@ def get_fields_to_fetch(export_type: ExportType):
     if export_type == "latest":
         return LATEST_LABEL_FIELDS
     return DEFAULT_FIELDS
+
+
+def is_geotiff_asset_with_lat_lon_coords(asset: Dict) -> bool:
```
Comment (PierreLeveau) [inline src/kili/services/export/tools.py:166]: should not be in this PR right?

File: src/kili/services/export/format/base.py [hunk 2]
```diff
@@ -229,6 +207,26 @@ def _has_data_connection(self) -> bool:
...
+                if self.label_format not in ("raw", "kili"):
+                    raise NotCompatibleOptions(
+                        "Cannot export geotiff assets with geospatial coordinates in"
+                        f" {self.label_format} format. Please use 'raw' or 'kili' format instead."
+                    )
+
+                if self.normalized_coordinates is False:
+                    raise NotCompatibleOptions(
+                        "Cannot export geotiff assets with geospatial coordinates with"
+                        " `normalized_coordinates=False`."
```
[the remark this answers is not in the document]
Comment (theodu) [reply src/kili/services/export/format/base.py:227]: @PierreLeveau how did we handle the case of resolutions update in the front for geotiff images ? do we currently support it ? do we send a corrupted value or do we prevent the front from updating the resolutions for geotiff images ?
If we supported geotiff resolutions, we could unormalized  the vertices.

**Review by theodu — CHANGES_REQUESTED** [review]
.

File: src/kili/services/export/format/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jonas1312) [reply src/kili/services/export/format/base.py:227]: > if all N assets are geotiff, and that the parameters are valid wrt the export, we will test these two conditions N times
> 
> ```
> if has_one_geotiff_asset:
>     #test the 2 conditions
> ```

yep, I changed it for this:

```python
has_geotiff_asset = any(is_geotiff_asset_with_lat_lon_coords(asset) for asset in assets)
if has_geotiff_asset:
	...
```

File: src/kili/services/export/media/image.py [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+        return image.size
+
+    # video with frames
+    if (
+        isinstance(asset["jsonContent"], list)
+        and len(asset["jsonContent"]) > 0
+        and Path(asset["jsonContent"][0]).is_file()
+    ):
+        image = Image.open(str(asset["jsonContent"][0]))
+        return image.size
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply src/kili/services/export/media/image.py:27]: > so here we suppose that we have already downloaded the image or the frames in the local machine right ?

first we look in the asset["resolution"], then local path yes (if user called kili.export_labels(with_assets=True)

File: src/kili/services/export/format/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PierreLeveau) [reply src/kili/services/export/format/base.py]: @theodu no specific treatment for geotiff images. We send the values, but they are not useful.

**Review by theodu — APPROVED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**